src/main/webapp/app/entities/competency/learning-path.model.ts

```diff
@@ -44,6 +44,28 @@ export interface LearningPathNavigationOverviewDto {
     learningObjects: LearningPathNavigationObjectDto[];
 }
 
+export interface CompetencyGraphNodeDto {
+    id: string;
+    label: string;
+    softDueDate: Date;
+    progress: number;
+    confidence: number;
+    mastery: number;
+    dimension?: NodeDimension;
+}
+
+export interface CompetencyGraphEdgeDto {
+    id: string;
+    source: string;
+    target: string;
+    relationType: CompetencyRelationType;
+}
+
+export interface CompetencyGraphDto {
```

Comment:
Regarding DTOs, I think we have the convention that we spell it `DTO` and not `Dto`